handler/handler.go

```diff
@@ -59,58 +56,122 @@ func (batchHandler BatchHandler) Handle(ctx context.Context, url string, events
 	return nil
 }
 
-func (bh BatchHandler) SendToES(ctx context.Context, esDestURL string, events []*models.SearchDataImportModel) error {
+// Preparing the payload and sending bulk events to elastic search.
+func (batchHandler BatchHandler) sendToES(ctx context.Context, esDestURL string, events []*models.SearchDataImportModel) error {
 
 	// Wait on semaphore if we've reached our concurrency limit
 	syncWaitGroup.Add(1)
```

Comment:
- Semaphore is removed as we are actually performing the action of data insertion at dp-elasticsearch library, so can be removed from here
- However, due to better performance, Waitgroup is useful for concurrent processessing for un-interrupted elastic search insertion for a given batch.

Reply:
There isn't any concurrent processing happening - there is one go function (new process started) but we force the other process to wait till this is completed and therefore block any new messages being consumed by the app.

Reply:
I guess if we want concurrent behaviour so the app consumes more messages while a batch is being processed then we will need to rework the code

Reply:
After looking into consumer about process of preparing/processing the batch for bulk update, I agree that there is not need for any concurrent handling in handler, hence removed both. However, there is no need for any further refactoring in consumer, as discussed, because there already exist a go routine handling concurrency while preparing the batch.